Hide premium themes on standalone Jetpack sites

mreishus opened the pull request:
#### Changes proposed in this Pull Request

* On non-atomic jetpack sites, do not allow premium themes to display in the theme showcase
* They're currently not supported
* When they do display, they can prompt the user to 'Upgrade to Activate' and to buy an expensive Jetpack plan. Even if they bought the plan, they would not be able to use calypso to install the theme
* By hiding the theme, we stop the user from going down a path of paying for something they cannot use

Example of free Jetpack site before change prompting to upgrade:
![2022-01-10_14-53](https://user-images.githubusercontent.com/937354/148843681-16375b54-cab2-4d4f-8c02-a348a5eb3ca6.png)

After change, no premium selector available even when the flag is on
![2022-01-10_15-43](https://user-images.githubusercontent.com/937354/148843949-489001c6-b574-4dda-9cc1-191358bb21c9.png)


#### Testing instructions

* Use JN or Docker to make a jetpack site connected to calypso
* Visit `calypso.localhost:3000/themes/<SITE>?flags=themes/premium` for simple, atomic and standalone jetpack sites
* You should be able to see premium themes on simple and atomic, but not on standalone jetpack

mreishus committed:
Hide premium themes on Jetpack sites

sixhours commented:
I think from the title I was expecting it to hide the premium themes themselves, but it looks like it's only supposed to hide the filter. Is that right?

I'm still seeing Skivers, Byrne, Payton, etc. in the Calypso theme showcase with my non-Atomic Jetpack site when using this branch, and using the flag:
<img width="1196" alt="Screen Shot 2022-01-11 at 11 46 52 AM" src="https://user-images.githubusercontent.com/2124984/148985257-903eeee5-0ee7-46c0-9a11-0bd9f303bfd0.png">
<img width="1492" alt="Screen Shot 2022-01-11 at 11 46 27 AM" src="https://user-images.githubusercontent.com/2124984/148985887-d0856a0f-9f6b-4a56-a068-5fe60d305646.png">

sixhours commented on client/my-sites/themes/themes-selection.jsx:
```diff
@@ -218,7 +224,7 @@ export const ConnectedThemesSelection = connect(
 			search,
 			page,
-			tier: config.isEnabled( 'themes/premium' ) ? tier : 'free',
+			tier: isSupportPremiumThemes ? tier : 'free',
```
Oh, maybe this should be hiding the themes. Let me check the value of `isSupportPremiumThemes` on my Jetpack site.

sixhours replied:
It's false, as expected.

mreishus replied:
Yes, this part was supposed to be hiding it. :thinking:

mreishus committed:
Centralize premium theme visibility logic, also apply to trending themes

mreishus committed:
Apply premium theme filtering to recommended themes

mreishus committed:
Undo isomorphic change

mreishus commented:
Centralized the logic and also applied to recommended and trending themes

mreishus commented:
Looking into unit test failures.

mreishus committed:
Trending/recommended themes: Only get siteId when state.ui exists

I was seeing this type of failure in unit tests:
```
  ● #getRecommendedThemes › should return empty themes list for unfetched filter

    TypeError: Cannot read properties of undefined (reading 'selectedSiteId')

       8 |  */
       9 | export default function getSelectedSiteId( state ) {
    > 10 |      return state.ui.selectedSiteId;
         |                      ^
      11 | }
      12 |
```

The unit tests were trying to get a selected site id, to check if the
site was a standlone jetpack site, but since state.ui does not exist,
the getSelectedSiteId function was crashing.

It seems a better fix is to change getSelectedSiteId to
state?.ui?.selectedSiteId, but it's such a core function to calypso I'm
weary of changing it.

sixhours reviewed (approved):
This is working to hide the themes on my Jetpack site in all views/search, but I can still see them on my WP.com sites. 🥳
